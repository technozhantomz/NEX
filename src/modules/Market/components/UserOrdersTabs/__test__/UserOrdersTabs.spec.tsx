import { render } from "@testing-library/react";
import counterpart from "counterpart";
import React from "react";

import { UsersOrdersTabs } from "../UserOrdersTabs";

jest.mock("counterpart", () => ({
  translate: jest.fn().mockImplementation((value) => {
    if (value === "pages.market.my_open_orders") {
      return "pages.market.my_open_orders";
    } else if (value === "pages.market.my_order_history") {
      return "pages.market.my_order_history";
    }
  }),
}));

jest.mock("../../UserOrdersTable", () => ({
  UserOrdersTable: jest.fn().mockReturnValue(<div>UserOrdersTable</div>),
}));

describe("UsersOrdersTabs component", () => {
  it("should render tabs with translated labels and UserOrdersTable", () => {
    const { getByText, asFragment } = render(<UsersOrdersTabs />);

    expect(asFragment()).toMatchSnapshot();

    const openOrdersTab = getByText("pages.market.my_open_orders");
    expect(openOrdersTab).toBeInTheDocument();

    const orderHistoryTab = getByText("pages.market.my_order_history");
    expect(orderHistoryTab).toBeInTheDocument();

    expect(counterpart.translate).toHaveBeenCalledWith(
      `pages.market.my_open_orders`
    );
    expect(counterpart.translate).toHaveBeenCalledWith(
      `pages.market.my_order_history`
    );
  });
});
