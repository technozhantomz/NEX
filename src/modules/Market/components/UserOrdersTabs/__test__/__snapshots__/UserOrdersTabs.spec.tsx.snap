// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersOrdersTabs component should render tabs with translated labels and UserOrdersTable 1`] = `
<DocumentFragment>
  <div
    class="ant-tabs ant-tabs-top ant-tabs-centered Tabs-sc-694db91e-0 sc-bcXHqe CYjqP eyzkhJ"
  >
    <div
      class="ant-tabs-nav"
      role="tablist"
    >
      <div
        class="ant-tabs-nav-wrap"
      >
        <div
          class="ant-tabs-nav-list"
          style="transform: translate(0px, 0px);"
        >
          <div
            class="ant-tabs-tab ant-tabs-tab-active"
            data-node-key="user-open-orders"
          >
            <div
              aria-controls="rc-tabs-test-panel-user-open-orders"
              aria-selected="true"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-user-open-orders"
              role="tab"
              tabindex="0"
            >
              pages.market.my_open_orders
            </div>
          </div>
          <div
            class="ant-tabs-tab"
            data-node-key="user-orders-history"
          >
            <div
              aria-controls="rc-tabs-test-panel-user-orders-history"
              aria-selected="false"
              class="ant-tabs-tab-btn"
              id="rc-tabs-test-tab-user-orders-history"
              role="tab"
              tabindex="0"
            >
              pages.market.my_order_history
            </div>
          </div>
          <div
            class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
          />
        </div>
      </div>
      <div
        class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
      >
        <button
          aria-controls="rc-tabs-test-more-popup"
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-hidden="true"
          class="ant-tabs-nav-more"
          id="rc-tabs-test-more"
          style="visibility: hidden; order: 1;"
          tabindex="-1"
          type="button"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="ant-tabs-content-holder"
    >
      <div
        class="ant-tabs-content ant-tabs-content-top"
      >
        <div
          aria-hidden="false"
          aria-labelledby="rc-tabs-test-tab-user-open-orders"
          class="ant-tabs-tabpane ant-tabs-tabpane-active"
          id="rc-tabs-test-panel-user-open-orders"
          role="tabpanel"
          tabindex="0"
        >
          <div
            class="sc-gswNZR eQnQWW"
          >
            <div>
              UserOrdersTable
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
